stats/opentelemetry/internal/pluginoption.go

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package internal defines the PluginOption interface.
+package internal
+
+import (
+	"google.golang.org/grpc/metadata"
+)
+
+// PluginOption is the interface which represents a plugin option for the
+// OpenTelemetry instrumentation component. This plugin option emits labels from
+// metadata and also creates metadata containing labels. These labels are
+// intended to be added to applicable OpenTelemetry metrics recorded in the
+// OpenTelemetry instrumentation component.
+//
+// This API is experimental. In the future, we hope to stabilize and expose this
+// API to allow plugins to inject labels of their choosing into metrics
+// recorded.
+type PluginOption interface {
+	// GetMetadata creates a MD with metadata exchange labels.
+	GetMetadata() metadata.MD
+	// GetLabels emits relevant labels from the metadata provided and the
+	// optional labels provided, alongside any relevant labels.
```

Comment:
Maybe mention where the metadata and labels come from?  The metadata is (all of?) the incoming RPC metadata, but I'm not sure about the labels.

Reply:
I purposely left this semantically vague because this is general purpose. To have it in one function call I pass in all the incoming RPC metadata, optional labels come from xDS (but can be arbitrary, locality etc) and the relevant labels come from local labels derived at plugin option init time. Thus, these three sources for any plugin option was my goal here.

Reply:
It's always going to be "incoming RPC metadata".  I don't think it makes sense to intentionally make this vague.

Reply:
Ok incoming RPC metadata is fixed, other two buckets aren't.